src/plugin/parse-json-results/parse-json-results-plugin.ts

```diff
@@ -32,30 +54,32 @@ export class ParseJSONResultsPlugin implements KyselyPlugin {
   ): Promise<QueryResult<UnknownRow>> {
     return {
       ...args.result,
-      rows: parseArray(args.result.rows),
+      rows: parseArray(args.result.rows, this.#objectStrategy),
     }
   }
 }
 
-function parseArray<T>(arr: T[]): T[] {
+function parseArray<T>(arr: T[], objectStrategy: ObjectStrategy): T[] {
+  const target = objectStrategy === 'create' ? [] : arr
```

Comment:
Should we use `Array.create({ length: arr.length })` instead of `[]`?

Reply:
Seems faster.. https://jsperf.app/fudaro

nice one!